From 7dcc67cb5909daa4fbb70d0600a86d0b84121410 Mon Sep 17 00:00:00 2001
From: Friedrich Vock <friedrich.vock@gmx.de>
Date: Thu, 20 Jun 2024 14:12:15 +0200
Subject: [PATCH 09/42] drm/amdgpu: Don't use doorbells for SDMA 5.2

There seems to be a race condition and they can cause hangs in rare
scenarios.

Signed-off-by: Friedrich Vock <friedrich.vock@gmx.de>
(cherry picked from commit 9264ec783c738b272d94a72542524223b65255ef)
(cherry picked from commit beec5cbd7260be325d8b20cc5314dfbd38332aae)
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
Signed-off-by: Matthew Schwartz <matthew.schwartz@linux.dev>
---
 drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c b/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
index bc9b240a3..6e65dfa22 100644
--- a/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
+++ b/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
@@ -1288,7 +1288,7 @@ static int sdma_v5_2_sw_init(void *handle)
 	for (i = 0; i < adev->sdma.num_instances; i++) {
 		ring = &adev->sdma.instance[i].ring;
 		ring->ring_obj = NULL;
-		ring->use_doorbell = true;
+		ring->use_doorbell = false;
 		ring->me = i;
 
 		DRM_INFO("use_doorbell being set to: [%s]\n",
-- 
2.47.1

